File: 155.java
import java.util.*;

class MinStack {

    Deque<Integer> itemStack, minStack;

    public MinStack() {
        itemStack = new ArrayDeque<>();
        minStack = new ArrayDeque<>();
    }

    public void push(int val) {
        itemStack.push(val);
        if (minStack.isEmpty()) {
            minStack.push(val);
        } else {
            minStack.push(Math.min(val, minStack.peek()));
        }
    }

    public void pop() {
        itemStack.pop();
        minStack.pop();
    }

    public int top() {
        return itemStack.peek();
    }

    public int getMin() {
        return minStack.peek();
    }
}

File: Python/155.py
class MinStack:

    def __init__(self):
        self.stack = []
        self.m_stack = []

    def push(self, val: int) -> None:
        self.stack.append(val)
        self.m_stack.append(min(self.m_stack[-1] if self.m_stack else val, val))

    def pop(self) -> None:
        self.m_stack.pop()
        self.stack.pop()

    def top(self) -> int:
        return self.stack[-1]

    def getMin(self) -> int:
        return self.m_stack[-1]
